py-polars/src/conversion/any_value.rs

```diff
@@ -371,51 +390,50 @@ pub(crate) fn py_object_to_any_value(ob: &PyAny, strict: bool) -> PyResult<AnyVa
             match &*type_name {
                 // Can't use pyo3::types::PyDateTime with abi3-py37 feature,
                 // so need this workaround instead of `isinstance(ob, datetime)`.
-                "date" => get_date,
-                "time" => get_time,
-                "datetime" => get_datetime,
-                "timedelta" => get_timedelta,
-                "Decimal" => get_decimal,
-                "range" => get_list,
+                "date" => get_date as InitFn,
+                "time" => get_time as InitFn,
+                "datetime" => get_datetime as InitFn,
+                "timedelta" => get_timedelta as InitFn,
+                "Decimal" => get_decimal as InitFn,
+                "range" => get_list as InitFn,
                 _ => {
                     // Support NumPy scalars.
                     if ob.extract::<i64>().is_ok() || ob.extract::<u64>().is_ok() {
-                        return get_int;
+                        return get_int as InitFn;
                     } else if ob.extract::<f64>().is_ok() {
-                        return get_float;
+                        return get_float as InitFn;
                     }
 
                     // Support custom subclasses of datetime/date.
                     let ancestors = ob.get_type().getattr(intern!(py, "__mro__")).unwrap();
                     let ancestors_str_iter = ancestors
                         .iter()
                         .unwrap()
-                        .map(|b| b.unwrap().str().unwrap().to_str().unwrap());
+                        .map(|b| b.unwrap().str().unwrap().to_string());
                     for c in ancestors_str_iter {
-                        match c {
+                        match &*c {
                             // datetime must be checked before date because
                             // Python datetime is an instance of date.
-                            "<class 'datetime.datetime'>" => return get_datetime,
-                            "<class 'datetime.date'>" => return get_date,
+                            "<class 'datetime.datetime'>" => return get_datetime as InitFn,
+                            "<class 'datetime.date'>" => return get_date as InitFn,
                             _ => (),
                         }
                     }
 
-                    get_object
+                    get_object as InitFn
                 },
             }
         }
     }
+    get_conversion_function(ob, ob.py())(ob, strict)
 
-    let type_object_ptr = PyType::as_type_ptr(ob.get_type()) as usize;
-
-    Python::with_gil(|py| {
-        LUT.with_gil(py, |lut| {
-            let convert_fn = lut
-                .entry(type_object_ptr)
-                .or_insert_with(|| get_conversion_function(ob, py));
+    // let type_object_ptr = ob.get_type().as_type_ptr() as usize;
 
-            convert_fn(ob, strict)
-        })
-    })
+    // Python::with_gil(|py| {
```

Comment:
@ritchie46 I can't figure out how to make this compile, maybe you have some ideas?

Reply:
Yes. See comments.